ui/src/experiment/ExperimentsRouter.js

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import {
+  EuiPage,
+  EuiPageBody,
+  EuiPageContent,
+  EuiPageHeader,
+  EuiPageHeaderSection,
+  EuiText,
+} from "@elastic/eui";
+import { PageTitle } from "../components/page/PageTitle";
+
+import useDynamicScript from "../hooks/useDynamicScript";
+import loadComponent from "../utils/remoteComponent";
+import { defaultExperimentEngine } from "../config";
+
+export const ExperimentsRouter = ({ projectId }) => {
+  // Retrieve script from host dynamically
+  const { ready, failed } = useDynamicScript({
+    url: defaultExperimentEngine.url,
```

Comment:
What if  `defaultExperimentEngine` is `undefined`?

Reply:
`defaultExperimentEngine` will at least be `{appName: undefined, url: undefined}`. Anyway, refactored this file based on your other comments, so this is no longer relevant.